src/Adapter/Currency/CommandHandler/ToggleExchangeRateAutomatizationHandler.php

```diff
@@ -0,0 +1,262 @@
+ * of period. If the setting is off then it removes that record.
+ *
+ * @todo: an issue with multi-store cron task scheduler.
+ *
+ * @internal
+ */
+final class ToggleExchangeRateAutomatizationHandler implements ToggleExchangeRateAutomatizationHandlerInterface
+{
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var Shop
+     */
+    private $contextShop;
+
+    /**
+     * @var bool
+     */
+    private $isCronJobModuleInstalled;
+
+    /**
+     * @var ShopUrlDataProvider
+     */
+    private $shopUrlDataProvider;
+
+    /**
+     * @var string
+     */
+    private $adminBaseUrl;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param Configuration $configuration
+     * @param Shop $contextShop
+     * @param ShopUrlDataProvider $shopUrlDataProvider
+     * @param TranslatorInterface $translator
+     * @param bool $isCronJobModuleInstalled
+     * @param string $adminBaseUrl
+     * @param string $dbPrefix
+     */
+    public function __construct(
+        Configuration $configuration,
+        Shop $contextShop,
+        ShopUrlDataProvider $shopUrlDataProvider,
+        TranslatorInterface $translator,
+        $isCronJobModuleInstalled,
+        $adminBaseUrl,
+        $dbPrefix
+    ) {
+        $this->configuration = $configuration;
+        $this->contextShop = $contextShop;
+        $this->isCronJobModuleInstalled = $isCronJobModuleInstalled;
+        $this->shopUrlDataProvider = $shopUrlDataProvider;
+        $this->adminBaseUrl = $adminBaseUrl;
+        $this->dbPrefix = $dbPrefix;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CurrencyException
+     */
+    public function handle(ToggleExchangeRateAutomatizationCommand $command)
+    {
+        if (!$this->isCronJobModuleInstalled) {
+            throw new ScheduleExchangeRatesUpdateException(
+                'Live exchange rates feature cannot be modified because "cronjob" module is not installed',
+                ScheduleExchangeRatesUpdateException::CRON_TASK_MANAGER_MODULE_NOT_INSTALLED
+            );
+        }
+
+        $this->configuration->restrictUpdatesTo($this->contextShop);
+
+        $cronId = (int) $this->configuration->get('PS_ACTIVE_CRONJOB_EXCHANGE_RATE');
+        $thereIsOneCronRunning = ($cronId !== 0);
+
+        try {
+            if ($thereIsOneCronRunning && $command->exchangeRateStatus()) {
+                $this->removeConfigurationIfNotFoundOrIsDeactivated($cronId);
+            }
+
+            if (!$thereIsOneCronRunning && $command->exchangeRateStatus()) {
+                $this->enableExchangeRatesScheduler();
+            }
+
+            if ($thereIsOneCronRunning && !$command->exchangeRateStatus()) {
+                $this->disableExchangeRatesScheduler($cronId);
+            }
```

Comment:
i didnt check it, but does it make sense to go to each `if`? let's say 2nd if is `true`, do you still need to check if 3rd `if` is `true`? in other words, shouldnt it be `if`/`else if` ?